drivers/pinctrl/pinctrl-bflb.c

```diff
@@ -0,0 +1,802 @@
+	kfree(girq->parents);
+	kfree(irq_data);
+
+	return ret;
+}
+
+static int bflb_gpio_pinctrl_probe(struct platform_device *pdev)
+{
+	struct bflb_gpio_pinctrl *pctl;
+	struct pinctrl_pin_desc *pins;
+
+	unsigned int npins;
+	const char **pin_names;
+	unsigned int *pin_nums;
+	unsigned int i, nirqs = 0;
+	int res;
+
+	if (of_property_read_bool(pdev->dev.of_node, "interrupt-controller")) {
+		res = platform_irq_count(pdev);
+		if (res > 0)
+			nirqs = res;
+	}
+
+	pctl = devm_kzalloc(&pdev->dev, sizeof(*pctl), GFP_KERNEL);
+
+	if (!pctl)
+		return -ENOMEM;
+
+	pctl->dev = &pdev->dev;
+	pctl->gpio_chip.irq.num_parents = nirqs;
+
+	dev_set_drvdata(&pdev->dev, pctl);
+
+	if (of_property_read_u32(pdev->dev.of_node, "bflb,npins", &npins))
+		return dev_err_probe(&pdev->dev, -EINVAL,
+				"bflb,npins property not found\n");
+
+	pins = devm_kmalloc_array(&pdev->dev, npins, sizeof(pins[0]), GFP_KERNEL);
+	pin_names = devm_kmalloc_array(&pdev->dev, npins, sizeof(pin_names[0]),
+			GFP_KERNEL);
+	pin_nums = devm_kmalloc_array(&pdev->dev, npins, sizeof(pin_nums[0]),
+			GFP_KERNEL);
+
+	if (!pins || !pin_names || !pin_nums)
+		return -ENOMEM;
+
+	pctl->base = devm_platform_ioremap_resource(pdev, 0);
+
+	if (IS_ERR(pctl->base))
+		return PTR_ERR(pctl->base);
+
+	pctl->map = devm_regmap_init_mmio(&pdev->dev, pctl->base, &regmap_config);
+
+	if (IS_ERR(pctl->map))
+		return dev_err_probe(&pdev->dev, PTR_ERR(pctl->map),
+				"Failed to create regmap\n");
+
+	for (i = 0; i < npins; i++) {
+		pins[i].number = i;
+		pins[i].name = devm_kasprintf(&pdev->dev, GFP_KERNEL, "GPIO%u", i);
+		pins[i].drv_data = pctl;
+		pin_names[i] = pins[i].name;
+		pin_nums[i] = i;
+	}
+
+	pctl->pinctrl_desc.name = dev_name(pctl->dev);
+	pctl->pinctrl_desc.pins = pins;
+	pctl->pinctrl_desc.npins = npins;
+	pctl->pinctrl_desc.pctlops = &bflb_gpio_pinctrl_ops;
+	pctl->pinctrl_desc.pmxops = &bflb_gpio_pinmux_ops;
+
+	pctl->pctldev =	devm_pinctrl_register(&pdev->dev, &pctl->pinctrl_desc,
+			pctl);
+
+	if (IS_ERR(pctl->pctldev))
+		return dev_err_probe(&pdev->dev, PTR_ERR(pctl->pctldev),
+				"Failed to register pinctrl device.\n");
+
+	for (i = 0; i < npins; i++) {
+		res = pinctrl_generic_add_group(pctl->pctldev,
+				pins[i].name, pin_nums + i, 1, pctl);
+
+		dev_dbg(&pdev->dev, "Registered pin %s with numeric %u",
+				pins[i].name, i);
+
+		if (res < 0)
+			return dev_err_probe(pctl->dev, res, "Failed to register group");
+	}
+
+	for (i = 0; i < ARRAY_SIZE(pinmux_functions); ++i) {
+		if (pinmux_functions[i]) {
+			res = pinmux_generic_add_function(pctl->pctldev,
+					pinmux_functions[i], pin_names, npins, pctl);
+
+			dev_dbg(&pdev->dev, "Registered function %s with numeric %u",
+					pinmux_functions[i], i);
+
+			if (res < 0)
+				return dev_err_probe(pctl->dev, res,
+						"Failed to register function.");
+		}
+	}
+
+	dev_info(&pdev->dev, "Bouffalo Lab pinctrl+GPIO(+interrupt) controller - \
+Registered %lu function(s) for %u pin(s)",
+
+			ARRAY_SIZE(pinmux_functions), npins);
```

Comment:
don't break strings with backslash.  Instead use two strings which the compiler will combine into one like this:

```
dev_info(&pdev->dev, "Bouffalo Lab pinctrl+GPIO(+interrupt) controller - "
                     "Registered %lu function(s) for %u pin(s)", 
                     ARRAY_SIZE(pinmux_functions), npins);
```